discovery-provider/src/utils/helpers.py

```diff
@@ -60,6 +60,16 @@ def model_to_dictionary(db_model_obj, exclude_keys=None, relationships_to_includ
     return model_dict
 
 
+# Convert a tuple of model format into the proper model itself.
+# The number of entries in the tuple, must map the modle
+def tuple_to_model_dictionary(t, model):
```

Comment:
I don't think so. The reason this is needed is because  I wanted to establish a subquery as the 'select_from' part in sql alchemy (rather than doing a filter against or join in that case).

Because we are basically doing
select * from (select * from playlists)

sql alchemy loses the concept of the model and just spits everything back as a tuple.



If you do this
```
subq = session.query(Playlist).subquery()

session.query(
  Playlist
).select_from(subq)

you end up with 

SELECT playlists.blockhash, 
       playlists.blocknumber, 
       playlists.playlist_id, 
       playlists.playlist_owner_id, 
       playlists.is_album, 
       playlists.is_private, 
       playlists.playlist_name, 
       playli sts.playlist_contents, 
       playlists.playlist_image_multihash, 
       playlists.playlist_image_sizes_multihash, 
       playlists.description, 
       playlists.upc, 
       playlists.is_current, 
       playlists.is_delete, 
       playlists.updated_at, 
       playlists.created_at 
FROM   playlists, 
       ( 
              SELECT playlists.blockhash                      AS blockhash, 
                     playlists.blocknumber                    AS blocknumber, 
                     playlists.playlist_id                    AS playlist_id, 
                     playlists.playlist_owner_id              AS playlist_owner_id, 
                     playlists.is_album                       AS is_album, 
                     playlists.is_private a s                    is_private, 
                     playlists.playlist_name                  AS playlist_name, 
                     playlists.playlist_contents              AS playlist_contents, 
                     playlists.playlist_image_multihash       AS playlist_image_multihash, 
                     playlists.playlist_image_sizes_multihash AS playlist_image_sizes_mul tihash, 
                     playlists.description                    AS description, 
                     playlists.upc                            AS upc, 
                     playlists.is_current                     AS is_current, 
                     playlists.is_delete                      AS is_delete, 
                     playlists.updated_at                     AS updated_at, 
                     playlists.created_at                     AS created_at 
              FROM   playlists) AS anon_1 limit 10

```

There might be a better way, but I couldn't come up with anything elegant. I updated the comment a bit more here.